src/domains/access.ml

```diff
@@ -263,20 +262,21 @@ let get_val_type e (vo: var_o) (oo: off_o) : acc_typ =
   with _ -> get_type voidType e
 
 let some_accesses = ref false
-let add_one (e:exp) (w:bool) (conf:int) (ty:acc_typ) (lv:(varinfo*offs) option) ((pp,lp):part): unit =
+let add_one (e:exp) (w:bool) (conf:int) (mhp:MHP.t) (ty:acc_typ) (lv:(varinfo*offs) option) ((pp,lp):part): unit =
   if is_ignorable lv then () else begin
+    Printf.printf "adding %s\n" (ThreadIdDomain.ThreadLifted.show mhp.tid);
     some_accesses := true;
     let tyh = TypeHash.find_def accs  ty (lazy (LvalOptHash.create 10)) in
     let lvh = LvalOptHash.find_def tyh lv (lazy (PartOptHash.create 10)) in
     let loc = !Tracing.current_loc in
     let add_part ls =
       PartOptHash.modify_def lvh (Some(ls)) (lazy (Set.empty,lp)) (fun (s,o_lp) ->
-          (Set.add (conf, w,loc,e,lp) s, LSSet.inter lp o_lp)
+          (Set.add (conf,mhp, w,loc,e,lp) s, LSSet.inter lp o_lp)
```

Comment:
Now this polymorphic `Set` contains some non-primitive data structures that make up `MHP.t`, including thread IDs in non-string forms. It would probably be safer to replace this `Set` with one constructed using `Set.Make`, which actually delegates to (currently non-existent) `MHP.compare`, etc. Using `[@@deriving ord]` hopefully makes this easy enough.

Reply:
Done!

Reply:
After working on #519 I remembered that that already seems to have a proper `SetDomain` for these anyway.